assets/src/edit-story/components/panels/preset/utils.js

```diff
@@ -63,15 +106,52 @@ export function generatePresetStyle(preset, prepareForCSS) {
   return style;
 }
 
-export function getTextPresets(elements, stylePresets) {
+function getExtractedInlineValue(value) {
+  return value !== MULTIPLE_VALUE ? value : null;
+}
+
+function getTextInlineStyles(content) {
+  const {
+    color,
+    fontWeight,
+    isBold,
+    isItalic,
+    isUnderline,
+    letterSpacing,
+  } = getHTMLInfo(content);
+  return {
+    color: color !== MULTIPLE_VALUE ? color : createSolid(0, 0, 0),
+    fontWeight: getExtractedInlineValue(fontWeight),
+    isBold: getExtractedInlineValue(isBold),
+    isItalic: getExtractedInlineValue(isItalic),
+    isUnderline: getExtractedInlineValue(isUnderline),
+    letterSpacing: getExtractedInlineValue(letterSpacing),
+  };
+}
+
+export function getTextPresets(elements, stylePresets, type) {
   // @todo Fix: Currently when two selected elements have the same attributes, two presets are added.
   return {
-    colors: elements
-      .map(({ content }) => getHTMLInfo(content).color)
-      .filter((color) => color !== MULTIPLE_VALUE)
-      .filter(
-        (color) => color && !findMatchingColor(color, stylePresets, true)
-      ),
+    colors:
+      'style' === type
+        ? []
+        : elements
+            .map(({ content }) => getHTMLInfo(content).color)
+            .filter((color) => color !== MULTIPLE_VALUE)
+            .filter(
+              (color) => color && !findMatchingColor(color, stylePresets, true)
+            ),
+    textStyles:
+      'color' === type
+        ? []
+        : elements
+            .map((text) => {
+              return {
+                ...objectPick(text, TEXT_PRESET_STYLES),
+                ...getTextInlineStyles(text.content),
+              };
+            })
+            .filter((preset) => !findMatchingStylePreset(preset, stylePresets)),
```

Comment:
This one isn't working correctly. If I add a text style for some configuration with a _fill_ background and then switch it to _highlight_ and try to add it again, I cannot. They're somehow interpreted to be identical, even though they have different background modes. Probably because the CSS at this level is identical, as the _highlight_ setting is added higher up normally?